src/azul/chalice.py

```diff
@@ -504,34 +497,105 @@ def swagger_resource(self, file) -> Response:
                                 headers={'Content-Type': content_type},
                                 body=body)
 
-    def threshold(self, *, errors: int, throttles: int):
-        def wrapper(f):
+    @attr.s(auto_attribs=True, frozen=True, kw_only=True)
```

Comment:
I believe the new convention is to use

```suggestion
    @attrs.define(frozen=True)
```